common/compiler/solidity.go

```diff
@@ -34,15 +34,10 @@ import (
 	"github.com/ethereum/go-ethereum/logger/glog"
 )
 
-const (
-	// flair           = "Christian <c@ethdev.com> and Lefteris <lefteris@ethdev.com> (c) 2014-2015"
-	flair           = ""
-	languageVersion = "0"
-)
-
 var (
-	versionRegExp = regexp.MustCompile("[0-9]+.[0-9]+.[0-9]+")
-	params        = []string{
+	versionRegExp = regexp.MustCompile("[0-9]+\\.[0-9]+\\.[0-9]+")
+	newAPIRegexp  = regexp.MustCompile("0\\.1\\.[2-9][0-9]*")
```

Comment:
@chriseth ok so what version matching should I use to catch legacy? I wanna be backward compatible on cpp versions but yeah it is also inconvenient to have to change the api when solc 0.2 is out. 
As an alternative you could include an API version in the output of --version so I can match on it from now on. Then I only have to match 0.1.2 additionally and the rest is legacy?